src/taxinyc/data_processor.py

```
@@ -0,0 +1,86 @@
+        Returns:
+            pd.DataFrame: Loaded data
+            
+        Raises:
+            ValueError: If sparksession is not active or table doesn't exist
+        """
+        if not sparksession:
+            raise ValueError("Invalid or inactive Spark session")
+        try:
+            return sparksession.read.table(table).toPandas()
+        except Exception as e:
+            raise ValueError(f"Failed to load table {table}: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)